server/e2e_test.go

```diff
@@ -0,0 +1,212 @@
+		relay.OverrideHandleGetPayload(func(w http.ResponseWriter, _ *http.Request) {
+			w.Header().Set("Content-Type", MediaTypeJSON)
+			w.WriteHeader(http.StatusBadRequest)
+			if _, err := w.Write([]byte(`{"code": 400, "message": "could not verify payload signature"}`)); err != nil {
+				t.Fatalf("failed to write response: %v", err)
+			}
+		})
+
+		reqHeader := http.Header{}
+		reqHeader.Set("Content-Type", "application/json")
+		resp := backend.request(t, http.MethodPost, params.PathGetPayload, reqHeader, signed)
+
+		require.Equal(t, http.StatusBadGateway, resp.Code)
+	})
+}
+
+func TestSignedBlindedBlockWithSlotMismatch(t *testing.T) {
+	backend, cleanup := setupTestBackend(t, 1, 500*time.Millisecond)
+	defer cleanup()
+
+	relay := backend.relays[0]
+	parentHash := mock.HexToHash(testBlockHash)
+	slotMismatch := testSlot + 1
+
+	relay.GetHeaderResponse = relay.MakeGetHeaderResponse(
+		10_000_000_000,
+		parentHash.String(),
+		parentHash.String(),
+		testPubKey,
+		spec.DataVersionElectra,
+	)
+
+	bid := setupBasicRequest(t, backend)
+	signed := createSignedBlindedBlock(bid.Electra.Message.Header, parentHash, slotMismatch, testSig)
+
+	reqHeader := http.Header{}
+	reqHeader.Set("Content-Type", "application/json")
+	resp := backend.request(t, http.MethodPost, params.PathGetPayload, reqHeader, signed)
+
+	require.Equal(t, http.StatusBadGateway, resp.Code)
+	require.Contains(t, strings.ToLower(resp.Body.String()), "no successful relay response")
+}
+
+func createSignedBlindedBlock(header *deneb.ExecutionPayloadHeader, blockHash phase0.Hash32, slot uint64, hexSig string) *eth2ApiV1Electra.SignedBlindedBeaconBlock {
+	sig := phase0.BLSSignature{}
+	if hexSig != "" {
+		sig = mock.HexToSignature(hexSig)
+	}
+	return &eth2ApiV1Electra.SignedBlindedBeaconBlock{
+		Signature: sig,
+		Message: &eth2ApiV1Electra.BlindedBeaconBlock{
+			Slot:          phase0.Slot(slot),
+			ProposerIndex: 1,
+			ParentRoot:    phase0.Root(blockHash),
+			StateRoot:     phase0.Root{0x01},
+			Body: &eth2ApiV1Electra.BlindedBeaconBlockBody{
+				RANDAOReveal: phase0.BLSSignature{0xaa},
+				ETH1Data: &phase0.ETH1Data{
+					BlockHash: blockHash[:],
+				},
+				Graffiti: phase0.Hash32{0xbb},
+				SyncAggregate: &altair.SyncAggregate{
+					SyncCommitteeBits: bitfield.NewBitvector512(),
+				},
+				ProposerSlashings:      []*phase0.ProposerSlashing{},
+				Deposits:               []*phase0.Deposit{},
+				VoluntaryExits:         []*phase0.SignedVoluntaryExit{},
+				ExecutionPayloadHeader: header,
+				AttesterSlashings:      []*electra.AttesterSlashing{},
+				Attestations:           []*electra.Attestation{},
+				BLSToExecutionChanges:  []*capella.SignedBLSToExecutionChange{},
+				BlobKZGCommitments:     []deneb.KZGCommitment{},
+				ExecutionRequests:      &electra.ExecutionRequests{},
+			},
+		},
+	}
+}
+
+func closeServers(relays []*mock.Relay) {
+	for _, relay := range relays {
+		relay.Server.Close()
+	}
+}
```

Comment:
Given that this is only used once, I think it would best to move it into `setupTestBackend`. Unless there's a reason this needs to be a separate function.